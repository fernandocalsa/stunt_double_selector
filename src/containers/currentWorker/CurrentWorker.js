import React, { Component } from 'react'
import { connect } from 'react-redux'
import WorkerCard from '../../components/WorkerCard'
import propTypes from './prop-types'

export class CurrentWorker extends Component {
  static propTypes = propTypes
  onHandleWorker = approved => {
    const { approve, reject, currentWorker } = this.props
    approved ? approve(currentWorker.id) : reject(currentWorker.id)
  }
  render() {
    const { currentWorker, load } = this.props
    if (!currentWorker) {
      return (
        <button onClick={load}>
          Load more actors
        </button>
      )
    }
    return (
      <WorkerCard worker={this.props.currentWorker} onHandleWorker={this.onHandleWorker} />
    )
  }
}

const mapState = ({ workers }) => ({
  currentWorker: workers.find(({ approved }) => approved === undefined)
})

const mapDispatch = ({ workers }) => ({
  approve: workers.approve,
  reject: workers.reject,
  load: workers.load
})

export default connect(mapState, mapDispatch)(CurrentWorker)
