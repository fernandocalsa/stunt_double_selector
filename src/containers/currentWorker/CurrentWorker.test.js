import React from 'react'
import { shallow } from 'enzyme'
import { CurrentWorker } from './CurrentWorker'

describe('CurrentWorker', () => {
  it('When there is a current worker it displays the worker card', () => {
    const props = {
      currentWorker: {
        id: 'testId',
        name: 'Test',
        imageUrl: 'test.jpg'
      },
      approve: jest.fn(),
      reject: jest.fn(),
      load: jest.fn()
    }
    const output = shallow(<CurrentWorker {...props} />)
    expect(output).toMatchSnapshot()
  })
  it('When there is no a current worker it displays a button', () => {
    const props = {
      approve: jest.fn(),
      reject: jest.fn(),
      load: jest.fn()
    }
    const output = shallow(<CurrentWorker {...props} />)
    expect(output).toMatchSnapshot()
  })
})