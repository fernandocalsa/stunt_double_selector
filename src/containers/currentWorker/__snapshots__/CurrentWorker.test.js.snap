// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrentWorker When there is a current worker it displays the worker card 1`] = `
<WorkerCard
  onHandleWorker={[Function]}
  worker={
    Object {
      "id": "testId",
      "imageUrl": "test.jpg",
      "name": "Test",
    }
  }
/>
`;

exports[`CurrentWorker When there is no a current worker it displays a button 1`] = `
<button
  onClick={[MockFunction]}
>
  Load more actors
</button>
`;
